src/main/java/frc/team997/lib/Trajectory/AsymmetricTrapezoidProfile.java

```diff
@@ -0,0 +1,355 @@
+             * constants, then we can use the quadratic formula to solve for accelTime
+             */
+            double vDiff = initialVelocity-targetVelocity;
+            double a = (0.5*maxAccel) - (Math.pow(maxAccel, 2)/(2*maxDecel));
+            double b = initialVelocity - ((initialVelocity*maxAccel)/maxDecel);
+            double c = -((Math.pow(vDiff, 2)/(2*maxDecel)) + ((targetVelocity*vDiff)/maxDecel) + targetPosition);
+            // subtracting the square root will always produce a negative result with positive target positions,
+            // and vice versa with negatives, so we can always add direction * sqrt.
+            accelTime = (-b + direction*Math.sqrt(Math.pow(b, 2)-(4*a*c)))/(2*a);
+            decelTime = -((maxAccel/maxDecel)*accelTime + vDiff/maxDecel);
+
+            // recalculate accel and decel position based off new times
+            accelPos = 0.5 * maxAccel * Math.pow(accelTime, 2) + accelTime*initialVelocity;
+            decelPos = -0.5 * maxDecel * Math.pow(decelTime, 2) + decelTime*targetVelocity;
+
+            // since we've adjusted the accel and decel we need to set the coast to 0
+            coastTime = 0;
+            coastPos = 0;
+
+            // if the deceleration time is negative, then our target velocity is higher than we can possibly get to
+            // it's probably best to just go as fast as we can and end in the right position
+            if (decelTime < 0) {
+                accelPos = targetPosition;
+                accelTime = (-initialVelocity + Math.sqrt(Math.pow(initialVelocity, 2) + 2*maxAccel*accelPos))/maxAccel;
+            }
+            // if the acceleration time is negative, then our target velocity is lower than we can possibly get to,
+            // and we should decelerate faster than our maximum deceleration so that we get to our position
+            // at the target velocity
+            if (accelTime < 0) {
+                decelPos = targetPosition;
+                decelTime = (2*decelPos)/(initialVelocity+targetVelocity);
+                maxDecel = (targetVelocity-initialVelocity)/decelTime;
+            }
+        }
+
+        Phase accelPhase = new Phase(accelTime, accelPos, maxAccel, initialVelocity);
+        Phase coastPhase = new Phase(coastTime, coastPos, 0, maxVelocity);
+        Phase decelPhase = new Phase(decelTime, decelPos, maxDecel, accelTime*maxAccel+initialVelocity);
+
+        if (accelPhase.time > 0)
+            phases.add(accelPhase);
+        if (coastPhase.time > 0)
+            phases.add(coastPhase);
+        if (decelPhase.time > 0)
+            phases.add(decelPhase);
+    }
+    /**
+     * Calculate the correct position and velocity for the profile at a given time
+     *
+     * @param time The time since the beginning of the profile.
+     * @return The position and velocity of the profile at the time.
+     */
+    public State calculate(double time) {
```

Comment:
I would call this _sample_ etc. because this isn't a controller (or, it's fully open-loop)

Reply:
that doc is just copy and pasted from the calculate function from wpilibs profile